src/Chat/ChatEditor.ts

```diff
@@ -41,13 +47,30 @@ export class ChatEditor {
     } else {
       chat.messages.push({ role: role, content: messageBlocks } as ChatMessage);
     }
+    this.addLanguageTag(messageBlocks, chat);
     this.chatDataManager.updateChat(chat);
 
     if (updateWebview) {
       await this.extensionToWebviewMessageSender.refresh();
     }
   }
 
+  /**
+   * Adds a language tag if appropriate
+   * @param messageBlocks most recent MessageBlocks appended to the chat.
+   * @param chat Chat to update.
+   */
+  public addLanguageTag(messageBlocks: MessageBlock[], chat: Chat) {
```

Comment:
Perhaps rename (and update docs) to `addLanguageTags` (plural)

Reply:
Renamed!